src/transformers/models/mamba/modeling_mamba.py

```diff
@@ -640,14 +681,25 @@ def prepare_inputs_for_generation(
         cache_params: Optional[MambaCache] = None,
         **kwargs,
     ):
+        use_cache = use_cache if use_cache is not None else (self.config.use_cache if not self.training else False)
+        if use_cache and self.training and self.gradient_checkpointing:
+            use_cache = False
+        if cache_params is None and use_cache:
+            # will construct a cache for a different generation
+            cache_params = MambaCache(
+                self.config,
+                input_ids.shape[0],
+                dtype=self.config.torch_dtype,
+                device=input_ids.device,
+            )
```

Comment:
I think we can initiate the cache instance which does not conform to the current standard inside
`prepare_inputs_for_generation`, users just have to pass in `use_cache=True` when using `generate` without the need of passing in specific cache implementation